src/Surface.zig

```diff
@@ -1336,16 +1334,12 @@ fn passwordInput(self: *Surface, v: bool) !void {
     try self.queueRender();
 }
 
-/// Sends a DSR response for the current color scheme to the pty. If
-/// force is false then we only send the response if the terminal mode
-/// 2031 is enabled.
-fn reportColorScheme(self: *Surface, force: bool) void {
-    if (!force) {
-        self.renderer_state.mutex.lock();
-        defer self.renderer_state.mutex.unlock();
-        if (!self.renderer_state.terminal.modes.get(.report_color_scheme)) {
-            return;
-        }
+/// Sends a DSR response for the current color scheme to the pty.
+fn reportColorScheme(self: *Surface) void {
+    self.renderer_state.mutex.lock();
+    defer self.renderer_state.mutex.unlock();
+    if (!self.renderer_state.terminal.modes.get(.report_color_scheme)) {
+        return;
     }
 
```

Comment:
Instead of duplicating this logic, we should add a new termio message so it's all sent from the same thread. Writes like this are already not synchronous ANYWAYS so that should be fine.